### PR TITLE
ACCUMULO-4835 Make TableOps throw TableNotFound

### PR DESCRIPTION
**Pull request by milleruntime**

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java [hunk 1]
```diff
@@ -1072,6 +1072,8 @@ private void waitForTableStateTransition(String tableId, TableState expectedStat
         if (Tables.getTableState(context.getInstance(), tableId) != expectedState) {
           if (!Tables.exists(context.getInstance(), tableId))
             throw new TableDeletedException(tableId);
+          if (Tables.getTableState(context.getInstance(), tableId) == TableState.DELETING)
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java]: After clearing the cache could store the return value of getTableState in a local var (could call it `currentState`).  Then could avoid two calls to getTableState after clearing the cache.

Comment (milleruntime) [reply core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java]: I thought about that but then was thinking that the state could change in that time.

Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java]: To properly account for changes between the times the two calls are made a loop would be needed.  However the wider code is in a loop, so its not needed here.

Comment (milleruntime) [reply core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java]: OK I gotcha.  I didn't fully grasp what this loop and method was doing in the first place.

File: core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java [hunk 2]
```diff
@@ -1069,14 +1069,17 @@ private void waitForTableStateTransition(String tableId, TableState expectedStat
...
           if (!Tables.exists(context.getInstance(), tableId))
             throw new TableDeletedException(tableId);
+          if (currentState == TableState.DELETING)
+            throw new TableNotFoundException(tableId, "", "Table is being deleted.");
           throw new AccumuloException("Unexpected table state " + tableId + " " + Tables.getTableState(context.getInstance(), tableId) + " != " + expectedState);
         }
       }
 
-      Range range = new KeyExtent(new Text(tableId), null, null).toMetadataRange();
+      Range range;
```
Comment (milleruntime) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java]: I also noticed this redundant assignment and removed it.

Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java]: Good catch

**Pull request merged**